src/main/java/com/koczy/kurek/mizera/thesisbrowser/service/DownloadService.java

```diff
@@ -0,0 +1,138 @@
+package com.koczy.kurek.mizera.thesisbrowser.service;
+
+import com.koczy.kurek.mizera.thesisbrowser.downloader.Parser.PdfParser;
+import com.koczy.kurek.mizera.thesisbrowser.downloader.PdfDownloader;
+import com.koczy.kurek.mizera.thesisbrowser.downloader.Scraper.AGHLibraryScraper;
+import com.koczy.kurek.mizera.thesisbrowser.downloader.Scraper.DblpScraper;
+import com.koczy.kurek.mizera.thesisbrowser.downloader.Scraper.GoogleScraper;
+import com.koczy.kurek.mizera.thesisbrowser.entity.Thesis;
+import com.koczy.kurek.mizera.thesisbrowser.model.ThesisFilters;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Service
+public class DownloadService implements IDownloadService {
+
+    private static final Logger logger = Logger.getLogger(DownloadService.class.getName());
+
+    private static final String REGEX = "[ :/*?|\"<>.]";
+    private static final String REPLACEMENT = "_";
+    private static final String TXT = ".txt";
+    private static final String PDF = ".pdf";
+
+    private AGHLibraryScraper aghLibraryScraper;
+    private DblpScraper dblpScraper;
+    private GoogleScraper googleScraper;
+    private PdfDownloader pdfDownloader;
+    private PdfParser pdfParser;
+
+    @Autowired
+    public DownloadService(AGHLibraryScraper aghLibraryScraper,
+                           DblpScraper dblpScraper,
+                           GoogleScraper googleScraper,
+                           PdfDownloader pdfDownloader,
+                           PdfParser pdfParser) {
+        this.aghLibraryScraper = aghLibraryScraper;
+        this.dblpScraper = dblpScraper;
+        this.googleScraper = googleScraper;
+        this.pdfDownloader = pdfDownloader;
+        this.pdfParser = pdfParser;
+    }
+
+    @Override
+    public ResponseEntity downloadTheses(ThesisFilters thesisFilters) {
+        String firstName = thesisFilters.getAuthor().split(" ")[0];
+        String lastName = thesisFilters.getAuthor().split(" ")[1];
+        Set<Thesis> theses = new HashSet<>();
+        try{
+            if(StringUtils.isEmpty(thesisFilters.getTitle())){
+                theses.addAll(findAllThesesFromAuthor(firstName, lastName));
+            }else{
+                theses.add(findThesisByAuthorNameAndTitle(firstName, lastName, thesisFilters.getTitle()));
+            }
+        } catch (IOException e) {
+            logger.log(Level.WARNING, e.toString());
+            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Wiem, że to ja dodałem, ale można by się zastanowić czy zwracamy taki status czy nie jakiś inny, plus można dopisać treść błędu jakąś od nas, new ResponseEntity("Jakis blad", HttpStatus.COS);